core/execution/market.go

```diff
@@ -2753,6 +2751,10 @@ func (m *Market) amendOrder(
 		// amended in place. Maybe a panic would be better
 		ret, err := m.orderAmendInPlace(existingOrder, amendedOrder)
 		if err == nil {
+			if sizeDecrease {
+				// ensure we release excess if party reduced the size of their order
+				m.recheckMargin(ctx, m.position.GetPositionsByParty(amendedOrder.Party))
+			}
```

Comment:
Is that necessary? Wouldn't the call at line 2718 be enough?

Reply:
No, it's not enough. That call only checks for insufficient margin, but does not release excess. The reason for this (IIRC) is the margin check is used when repricing LP's and adding new orders. Releasing excess - especially when repricing LP's - leads to a lot of redundant transfers, and potentially excessive bond slashing (release -> bond account -> place next order -> take from bond + slash, etc...)

Reply:
hum interesting.